automation.md

```diff
@@ -0,0 +1,102 @@
+![background](img/logo.png)
+
+[Clicknium](https://www.clicknium.com/) offers automation technology for a variety of uses. It can handle several forms of automation, including SAP automation, Java automation, Windows desktop automation, and image automation. Let's discuss about the automation technology that underlies.
+
+# Introduction
+
+Regarding software automation technology, it was mainly used for software testing at the earliest, especially when it comes to UI automation testing. Many system software designs take into account how to make it accessible to people with disabilities. See [windows accessibility](https://www.microsoft.com/en-us/accessibility/?rtc=1). In recent years, RPA (Robotic Process Automation) has been accepted by more and more enterprises. RPA can play a great role in multiple business field scenarios. process, to reduce costs and increase efficiency for enterprises, and quickly improve the level of informatization. RPA products generally include automated operation components for various software, data processing components, process logic control components, etc., of which UI automation components are the most used.
+
+# Windows UI Automation Technology
+
+Windows applications include such a functional design: Accessibility, how to allow more people to operate and use the software, such as how to allow the blind to use, read the screen through voice, etc. The UI automation technology provided by Microsoft can access the control elements of the windows application. The main technologies are as follows:
+- UIA (UI Automation) 
+UI Automation provides programmatic access to information about the user interface (UI) for Microsoft windows, enabling assistive technology products (such as screen readers) to provide information about the UI to end users using means other than standard input and Manipulate UI. UI Automation also enables automated test scripts to interact with the UI.
+- MSAA（Microsoft Active Accessibility）
+Microsoft Active Accessibility was the earlier application accessibility solution from Microsoft. 
+
+UIA is the new accessibility model for Microsoft Windows, offers many improvements over MSAA.
+
+||UIA|MSAA|
+|--------|---------|-------|
+|Programing Languages| written in managed code, client applications are most easily programmed using C# | based on the Component Object Model (COM) with support for dual interfaces.|
+|Windows Presentation Foundation|full support |WPF do not contain native support for MSAA|
+|Servers and Clients|a core service lies between the server (called a provider) and the client, expose more information to client|servers and clients communicate directly, largely through the server's implementation of IAccessible|
```

Comment:
|Servers and Clients|a core service between the server (called a provider) and the client, providing more information to the client|servers and clients communicate directly, largely through the server's implementation of IAccessible|